engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -0,0 +1,2532 @@
+            } else {
+              addend = digit;
+            }
+            sum += addend;
+            timesTwo = !timesTwo;
+          }
+
+          int modulus = sum % 10;
+          if ( modulus == 0 ) {
+            returnCode = true;
+          }
+        } catch ( Exception e ) {
+          // No Need to throw exception
+          // This means that input can not be parsed to Integer
+        }
+
+      }
+    } else {
+      throw new RuntimeException( "The function call LuhnCheck requires 1 argument." );
+
+    }
+    return returnCode;
+  }
+
+  public static int indexOf( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+
+    int returnIndex = -1;
+
+    if ( ArgList.length == 2 || ArgList.length == 3 ) {
+      if ( !isNull( ArgList ) && !isUndefined( ArgList ) ) {
+        String string = (String) ArgList[0];
+        String subString = (String) ArgList[1];
+
+        int fromIndex = 0;
+        if ( ArgList.length == 3 ) {
+          fromIndex = (int) Math.round( (Double) ArgList[2] );
+        }
+        returnIndex = string.indexOf( subString, fromIndex );
+      }
+    } else {
+      throw new RuntimeException( "The function call indexOf requires 2 or 3 arguments" );
+    }
+    return returnIndex;
+  }
+
+  public static Object getTransformationName( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    try {
+      Object objTranName = actualObject.get( "_TransformationName_" );
+      return objTranName;
+    } catch ( Exception e ) {
+      throw new RuntimeException( e.toString() );
+    }
+  }
+
+  public static void appendToFile( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+
+    if ( !isNull( ArgList ) && !isUndefined( ArgList ) ) {
+      try {
+        FileOutputStream file = new FileOutputStream( (String) ArgList[0], true );
+        DataOutputStream out = new DataOutputStream( file );
+        out.writeBytes( (String) ArgList[1] );
+        out.flush();
+        out.close();
+      } catch ( Exception er ) {
+        throw new RuntimeException( er.toString() );
+      }
+    } else {
+      throw new RuntimeException( "The function call appendToFile requires arguments." );
+    }
+  }
+
+  public static Object getFiscalDate( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+
+    if ( ArgList.length == 2 ) {
+      try {
+        if ( isNull( ArgList ) ) {
+          return null;
+        } else if ( isUndefined( ArgList ) ) {
+          return undefinedValue;
+        }
+        java.util.Date dIn = (java.util.Date) ArgList[0];
+        Calendar startDate = Calendar.getInstance();
+        Calendar fisStartDate = Calendar.getInstance();
+        Calendar fisOffsetDate = Calendar.getInstance();
+        startDate.setTime( dIn );
+        Format dfFormatter = new SimpleDateFormat( "dd.MM.yyyy" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.